sockeye/translate.py

```diff
@@ -67,40 +68,60 @@ def main():
                                                   *sockeye.inference.load_models(context,
                                                                                  args.max_input_len,
                                                                                  args.beam_size,
+                                                                                 args.batch_size,
                                                                                  args.models,
                                                                                  args.checkpoints,
                                                                                  args.softmax_temperature,
                                                                                  args.max_output_length_num_stds))
-        read_and_translate(translator, output_handler, args.input)
+
+        logger.info("Using batches of size %d", args.batch_size)
+        read_and_translate(translator, output_handler, args.portion_size, args.input)
 
 
 def read_and_translate(translator: sockeye.inference.Translator, output_handler: sockeye.output_handler.OutputHandler,
-                       source: Optional[str] = None) -> None:
+                       portion_size: int, source: Optional[str] = None) -> None:
     """
     Reads from either a file or stdin and translates each line, calling the output_handler with the result.
 
     :param output_handler: Handler that will write output to a stream.
     :param translator: Translator that will translate each line of input.
+    :param portion_size: The size of the portion to read at a time from the input
     :param source: Path to file which will be translated line-by-line if included, if none use stdin.
     """
+    def grouper(iterable: Iterable, size: int) -> Iterable:
+        """
+        Collect data into fixed-length chunks or blocks
+        without either dicarding underfilled chunks or padding them
+        """
+        it = iter(iterable)
+        while True:
+            chunk = list(itertools.islice(it, size))
+            if not chunk:
+                return
+            yield chunk
 
     source_data = sys.stdin if source is None else sockeye.data_io.smart_open(source)
 
     logger.info("Translating...")
 
-    i, total_time = translate_lines(output_handler, source_data, translator)
+    total_time, total_lines = 0.0, 0
+    for portion in grouper(source_data, portion_size):
+        lines, portion_time = translate(output_handler, portion, translator)
+        total_lines += lines
+        total_time += portion_time
 
-    if i != 0:
-        logger.info("Processed %d lines. Total time: %.4f sec/sent: %.4f sent/sec: %.4f", i, total_time,
-                    total_time / i, i / total_time)
+    if total_lines != 0:
+        logger.info("Processed %d lines in %d batches. Total time: %.4f, sec/sent: %.4f, sent/sec: %.4f",
+                    total_lines, -(-lines // translator.batch_size), total_time,
```

Comment:
why the double minus?

Reply:
A shorter equivalent of math.ceil(lines // translator,batch_size), will change to math.ceil.